spec/policies/course_policy_spec.rb

```diff
@@ -25,4 +25,30 @@
       it { should_not permit(other_user, course) }
     end
   end
+
+  describe CoursePolicy::TrainingProviderCourses do
+    let(:accredited_body) { create(:provider, :accredited_body, organisations: [organisation]) }
+    let(:training_provider) { create(:provider) }
+    let!(:course) { create(:course, provider: training_provider, accrediting_provider: accredited_body) }
+    let!(:other_course) { create(:course) }
+
+    subject { described_class.new(user, Course).resolve }
+
+    before { organisation }
+
+    context "user from the accredited_body" do
+      it { is_expected.to contain_exactly(course) }
+    end
+
+    context "user not from the accredited body" do
+      let(:organisation) { create(:organisation) }
+
+      it { is_expected.to be_empty }
+    end
+
+    context "an admin" do
+      let(:user) { create(:user, :admin) }
+      it { is_expected.to contain_exactly(course, other_course) }
```

Comment:
should the other_course be returned?
as the course is not from `accredited_body`
or from the `training_provider`

given its `CoursePolicy::TrainingProviderCourses`

Reply:
I don't think I can do that in a scope as it only has the user and the input scope to go on. For an admin user we'd have no reference to what the accredited body is, or the training provider. We're reliant on the input scope doing that and this policy scope just ensuring that if the user isn't an admin the courses are filtered appropriately.